### PR TITLE
#23 - Add code coverage support

### LINKED ISSUE #23: Add code coverage support
## Background
Add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- added new plugin dependency
- extended readme.md file with how to get code coverage report
- extended build logic to add JaCoCo non-default settings

Closes #23

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -34,6 +34,37 @@ ThisBuild / printScalaVersion := {
   log.info(s"Local maven ${Resolver.mavenLocal}")
 }
 
+lazy val jacocoCoreReportSettings = Seq(
```
Comment (Zejnilovic) [inline build.sbt:37]: Same feedback as in atum https://github.com/AbsaOSS/atum/pull/144#discussion_r1021678328

Comment (miroslavpojer) [reply build.sbt]: Resolved by commit 58f921a73167354e524e63072311893ae5b23b94.

File: build.sbt [hunk 2]
```diff
@@ -34,35 +35,13 @@ ThisBuild / printScalaVersion := {
...
-
-lazy val jacocoSlickReportSettings = Seq(
-  jacocoReportSettings := JacocoReportSettings(
-    s"fa-db:slick Jacoco Report",
-    None,
-    JacocoThresholds(),
-    Seq(JacocoReportFormats.HTML, JacocoReportFormats.XML),
-    "utf-8")
+lazy val commonJacocoReportSettings: JacocoReportSettings = JacocoReportSettings(
+  formats = Seq(JacocoReportFormats.HTML, JacocoReportFormats.XML)
```
Comment (benedeki) [inline build.sbt:39]: Do I guess correctly that the other settings, like `JacocoThresholds()` that were omitted are default?

File: build.sbt [hunk 3]
```diff
@@ -80,15 +59,23 @@ lazy val faDbCore = (project in file("core"))
     libraryDependencies ++= coreDependencies(scalaVersion.value),
     (Compile / compile) := ((Compile / compile) dependsOn printScalaVersion).value // printScalaVersion is run with compile
   )
-  .settings(jacocoCoreReportSettings: _*)
+  .settings(
+    jacocoReportSettings := commonJacocoReportSettings.withTitle("fa-db:core Jacoco Report"),
+    jacocoExcludes := commonJacocoExcludes ++ Seq(
+//      "za.co.absa.fadb.naming_conventions.AsIsNaming" // extra exclude example
```
Comment (benedeki) [inline build.sbt:65]: This one, I would leave out.
Examples were given above, how to do them. And extending common `val Seq` this way is standard approach in _sbt_

File: build.sbt (hunk 2 above)
Comment (miroslavpojer) [reply build.sbt:39]: Yes. I have removed default constructor parameters.

File: build.sbt (hunk 3 above)
Comment (miroslavpojer) [reply build.sbt]: Additional example of exclude removed in commit 722618e324218cb45457c35a47959a37cfaffd00.

**Review by benedeki — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run (partially)

**Pull request merged**